### ISSUE TITLE
Adding extensions to the WireMock when deployed as a war

### ISSUE THREAD

**Issue by nishantmc** [tomakehurst/wiremock#378]
I was trying wiremock using the sample war provided as part of the codebase. The mappings worked fine , but when i tried adding extensions to have dynamic responses, I am not sure where to add these classes to the container. I understood that along with the stub mappings i can provide a list of transformers just like this 

`    "response": {
        "status": 200,
        "bodyFileName": "response.json",
        "headers": {
            "Content-Type": "application/json"
        },
        "transformers": ["cutomer-create"]
    }`

But how the container will load the response classes. Then was going through each step in which wiremock app gets created, i could find that in WireMockWebContextListener, the WireMockApp is created by passing a list of empty transformers, i had to extend this listener and recreate the whole context initialize function to make the transformers working. Just want to check whether its a know issue or i am missing something.

**Comment by nishantmc** [comment]
This is how i added the transformers in the WireMockWebContextListener.
```
        Map<String, ResponseTransformer> transformers = new HashMap<String, ResponseTransformer>();
        transformers.put("cus-search", new CusSearchRespTransformer());

        WireMockApp wireMockApp = new WireMockApp(
                new NotImplementedRequestDelayControl(),
                false,
                defaultMappingsLoader,
                mappingsSaver,
                false,
                maxRequestJournalEntries,
                transformers,
                fileSource,
                new NotImplementedContainer()
        );

```

**Comment by tomakehurst** [comment]
Thanks for sharing your solution. Presume this is your own copy of the context listener?

I have to admit, WAR support is a bit neglected at the moment. This is the first indication I've had in quite a while that anyone still uses it.

**Comment by nishantmc** [comment]
yes its my own copy.  Thanks for the response. Let me add this hack to keep myself moving.

**Comment by bPopovska** [comment]
Is there a way to register a response transformer in a set up alike this one https://github.com/tomakehurst/wiremock/tree/master/sample-war i.e. everything is configured in web.xml ?

**Comment by tomakehurst** [comment]
@bPopovska currently not unfortunately

**Comment by Ntcha** [comment]
Another way to solve it to extend the WarConfiguration class and override the extensionsOfType method :
` private class CAPOptions extends WarConfiguration {
        public CAPOptions(ServletContext servletContext) {
            super(servletContext);
        }

        @Override
        public <T extends Extension> Map<String, T> extensionsOfType(final Class<T> extensionType) {
            Map<String, T> extensions = new LinkedHashMap<String, T>();
            Extension capTransformer = new CAPTransformer();
            extensions.put(capTransformer.getName(),(T) capTransformer);

           if (extensionType.isAssignableFrom(capTransformer.getClass())) {
               return extensions;
            }
            return Collections.emptyMap();

        }
    }`

That's obviously dirty but makes the job
